opendatahub-io/odh-dashboard
# [RHOAIENG-4702] Improve unique naming for runs and versions in pipelines

Closes: [RHOAIENG-4702](https://issues.redhat.com/browse/RHOAIENG-4702)

## Description
- Added duplicate name info helper text when uploading a pipeline, version or creating a run or schedule.
- Fixed issue where if > 20 versions existed within a particular pipeline, only the latest 20 were being used in the list of versions used on every runs table - this would result in pipeline versions not rendered in the table - instead just a hyphen, and consequently effected the list of pipeline versions available for the table filter toolbar dropdown.
- Fixed duplicate running of ~30 tests that was leading to an unnecessary additional ~2.5 minutes of CI cypress test runtime.

<img width="630" alt="image" src="https://github.com/user-attachments/assets/f074631d-2efc-4260-ae73-01d9c3cc3b1f">

## How Has This Been Tested?
Manual testing, and updated existing tests to verify the existence of the duplicate name helper text.

## Test Impact
Creation of runs, schedules, and importing of pipelines and versions.

## How to test
1. When creating a new run, specify a "Name" of a run that already exists within the project you're in, verify Info helper text exists, with the ability to still create the run, since this is not an API limitation for runs or recurring runs.
2. Perform step 1, but instead use the "Create schedule" page/form.
3. Open the import pipeline modal from the pipelines list page, specify a "Name" of a pipeline that already exists within the project you're in, verify helper text exists.
4. Perform step 3 except from the "Create run" page (accessed from below the Pipeline form dropdown).
5. Open the import pipeline version modal from the pipelines list page, specify a "Version name" of a pipeline version that already exists within the project you're in and the pipeline specified above it, verify helper text exists.
6. Perform step 5 except from the "Create run" page (accessed from below the Pipeline version form dropdown).

Self checklist (all need to be checked):

If you have UI changes: 

After the PR is posted & before it merges:
